app/controllers/application_controller.rb

```diff
@@ -11,6 +12,24 @@ def cors_preflight
 
   private
 
+  rescue_from "Exception" do |exception|
+    log_error(exception)
+    case exception
+    when VA::API::Common::Exceptions::BaseError
+      render json: { errors: exception.errors }
+    else
+      render json: { errors: VA::API::Common::Exceptions::InternalServerError.new(exception).errors }
+      # FIXME: do we need to re-raise the exception as long as we are logging it and rendering something?
+      # Maybe we will need a new relic specific method to call here??
+      # raise exception if Rails.env.production?
+    end
+  end
+
+  def log_error(exception)
+    Rails.logger.error "#{exception.message}."
+    Rails.logger.error exception.backtrace.join("\n") unless exception.backtrace.nil?
```

Comment:
@saneshark I think we're going to have a meeting later this week re:logging and how it will work with cloudwatch; but the backtrace could be part of the first log line (have to handle nil backtrace somehow):
```ruby
Rails.logger.error "#{exception.message}.", backtrace: exception.backtrace
```
which would output:
```javascript
{
	"time": "2016-08-30 22:19:57",
	"type": "ERROR",
	"msg": "something bad.",
	"backtrace": [
	  "/Users/alastair/Documents/adhoc/va/vets-api/lib/mvi/messages/find_candidate_message.rb:21:in `find_candidate_body'", 
	  "/Users/alastair/Documents/adhoc/va/vets-api/lib/mvi/messages/find_candidate_message.rb:13:in `build'", 
	  "/Users/alastair/Documents/adhoc/va/vets-api/spec/lib/mvi/messages/find_candidate_message_spec.rb:7:in `block (4 levels) in \u003ctop (required)\u003e'", 
	  "/Users/alastair/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/memoized_helpers.rb:295:in `block (2 levels) in let'", 
	  "/Users/alastair/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.2/lib/rspec/core/memoized_helpers.rb:157:in `block (3 levels) in fetch_or_store'",
	  // ...
}
```
We could also write some custom logic in the logger.error method so you could just pass in the exception and it would place stuff under the correct key.